pkg/lightning/backend/tidb/tidb.go

```diff
@@ -287,6 +296,22 @@ func (*targetInfoGetter) CheckRequirements(ctx context.Context, _ *backend.Check
 	return nil
 }
 
+// stmtKey defines key for stmtCache.
+type stmtKey struct {
+	query string
+	// `hash` is the hash value of this object.
+	hash []byte
+}
+
+// Hash implements SimpleLRUCache.Key.
+func (k *stmtKey) Hash() []byte {
+	if len(k.hash) == 0 {
+		k.hash = make([]byte, 0, len(k.query))
```

Comment:
Maybe use table name + number of rows as key? I'm not sure if comparing the bytes is a bottleneck in lightning's CPU profile.